store.go

```diff
@@ -30,3 +30,50 @@ type Store interface {
 	// Delete removes credentials from the store for the given server address.
 	Delete(ctx context.Context, serverAddress string) error
 }
+
+// storeWithFallbacks is a store that has multiple fallback stores.
+// Please use the NewStoreWithFallbacks to create new instances of
+// storeWithFallbacks.
+type storeWithFallbacks struct {
+	stores []Store
+}
+
+// NewStoreWithFallbacks returns a new store based on the given stores.
+// The first store is used as the primary store. The second and the
+// subsequent stores will be used as fallbacks for the first store.
+func NewStoreWithFallbacks(store Store, fallbacks ...Store) Store {
+	if fallbacks == nil {
```

Comment:
```suggestion
	if len(fallbacks) == 0 {
```
`fallbacks` can be not-nil but empty.

Reply:
changed.